core/src/main/resources/reference.conf

```diff
@@ -102,6 +102,32 @@ datastax-java-driver {
   # and metrics.
   // session-name = my_session
 
+  # A set of customizable actions that are performed when the driver starts up, i.e.,
+  # each time a new session is initialized.
+  # Actions in this section that involve printing log messages will always use the special logger
+  # 'com.datastax.oss.driver.api.core.Startup' at level INFO.
+  startup {
+
+    # Whether or not to print basic information about the initialization process,
+    # such as the driver name, its version number, and the process PID, if available.
+    print-basic-info = true
```

Comment:
Why not combine the two options? We could have a single `startup-log` that defaults to the basic info, and then you can customize it, or comment it out to disable the log.

Artifact coordinates can't use regular interpolation, but we could come up with a custom one and use a simple search and replace:
```
startup-log = "Starting #{driver.groupId}:#{driver.artifactId}:#{driver.version}"
```

Reply:
1. I like the idea of combining everything into one banner.
2. I don't like mixing two interpolation levels, I think users will get confused.

I did try today to come up with a solution entirely handled internally by TypeSafe Config, and using the new `ConfigResolver` API. It looked very promising, e.g. the following could have been the default banner:

```
banner = ${oss-driver.name} (${oss-driver.groupId}":"${oss-driver.artifactId}) version ${oss-driver.version} (PID ${pid})
```

But unfortunately it seems that there is a bug (or feature) in this new API, see https://github.com/lightbend/config/issues/553.

In the meanwhile, I'm having second thoughts about these banners. An even better solution would be to implement a listener interface for `Session` lifecycle events (`onSessionInitializing`, `onSessionInitialized`, `onSessionClosing`, etc.), and provide by default an implementation that logs basic driver info + time elapsed when the session is fully initialized.

Unless anyone disagrees, I will implement the idea of a listener.

Reply:
It turns out the bug is a feature :) so I suggest that we abandon the idea of a custom banner in favor of a listener.